lib/zlib.js

```diff
@@ -148,157 +147,220 @@ function flushCallback(level, strategy, callback) {
 // This thing manages the queue of requests, and returns
 // true or false if there is anything in the queue when
 // you call the .write() method.
-function Zlib(opts, mode) {
-  Transform.call(this, opts);
-  var chunkSize = Z_DEFAULT_CHUNK;
-  var flush = Z_NO_FLUSH;
-  var finishFlush = Z_FINISH;
-  var windowBits = Z_DEFAULT_WINDOWBITS;
-  var level = Z_DEFAULT_COMPRESSION;
-  var memLevel = Z_DEFAULT_MEMLEVEL;
-  var strategy = Z_DEFAULT_STRATEGY;
-  var dictionary;
-  if (opts) {
-    chunkSize = opts.chunkSize;
-    if (chunkSize !== undefined && chunkSize === chunkSize) {
-      if (chunkSize < Z_MIN_CHUNK || !Number.isFinite(chunkSize))
-        throw new RangeError('Invalid chunk size: ' + chunkSize);
-    } else {
-      chunkSize = Z_DEFAULT_CHUNK;
-    }
+class Zlib {
```

Comment:
This is a breaking API change that we tried to make previously. Unfortunate it breaks far too much of the ecosystem for us to do this. The fundamental reason it is breaking is that it breaks anyone who is inheriting from the `Zlib` class.